deploy/charts/cert-manager/README.template.md

```diff
@@ -148,6 +148,9 @@ The following table lists the configurable parameters of the cert-manager chart
 | `webhook.securityContext` | Security context for webhook pod assignment | `{}` |
 | `webhook.containerSecurityContext` | Security context to be set on the webhook component container | `{}` |
 | `webhook.hostNetwork` | If `true`, run the Webhook on the host network. | `false` |
+| `webhook.serviceType` | The type of the `Service`. | `ClusterIP` |
+| `webhook.loadBalancerIP` | The specific load balancer IP to use (when `serviceType` is `LoadBalancer`). |  |
+| `webhook.urlHost` | The host to use to reach the webhook, instead of using internal cluster DNS for the service. |  |
```

Comment:
Digging in, it seems like we force `https://` at the start here - we should maybe document an example of a url here to make this clear for users (I don't think _any other_ schemes are acceptable?). Otherwise, we could *not* force `https://` and then if a user does omit it or attempt anything else, presumably k8s API validation would fail?

Reply:
You're right, only HTTPS is allowed according to the [Kubernetes documentation](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#url):
> The scheme must be "https"; the URL must begin with "https://".

The idea behind just having a `urlHost` (i.e. `cert-manager-webhook.office.local`) was that it can be used both to build the `url` for the webhook configurations **and** for the certificate SAN in the deployment. But maybe it's better to completely separate them, and let people configure the full URLs?

```yaml
webhook:
  additionalCertificateSANs: cert-manager-webhook.office.local
  validateURL: https://cert-manager-webhook.office.local/validate
  mutateURL: https://cert-manager-webhook.office.local/mutate
```

It doesn't feel great having to configure `/validate` and `/mutate` though, as it's something that can't really change.

Or keep it as it is and improve documentation. Thoughts?

Reply:
I think you make a great point here re the certificate SAN too (and also the `/validate` and `/mutate` endpoint). What about making this `webhook.url.host` (i.e. dot-separating the url and host portion) and then keeping it as is? That will allow us in future to add `path` if that does ever make sense, and also keeps things nice and simply. Also by making it `url.host`, in my mind at least, this reads more like we just want the *host portion* of the URL (which is precisely what we want 😄).

Building on that.. do we want `webhook.url.port` too? We can probably skip it for now I guess (until there's an actual user request for it)

Reply:
Very valid points! I'll go with `webhook.url.host`. While `webhook.url.port` also makes sense, I'll leave it for now, as it feels YAGNI for my case. :)